File: go/samples/builder/from_file/from_file.go
package main

import (
	"errors"
	"io/ioutil"

	"github.com/bloock/bloock-sdk-go/v2/builder"
	"github.com/bloock/bloock-sdk-quickstart/utils"
	"github.com/fatih/color"
)

func main() {
	utils.Sample("builder_from_file", func(c utils.Config) error {
		file, err := ioutil.ReadFile("resources/dummy.pdf")
		if err != nil {
			return err
		}

		record, err := builder.NewRecordBuilderFromFile(file).Build()
		if err != nil {
			return err
		}

		color.Green("[✓] Record was created successfully")

		hash, err := record.GetHash()
		if err != nil {
			return err
		}

		if hash != "43fa336d95e1634fee2d3031adc44ed9464472b94511af1facb87a1fee0b7e0e" {
			return errors.New("Unexpected hash received")
		}

		color.Green("[✓] %s", hash)

		// we can get the file back if needed
		err = ioutil.WriteFile("resources/output.pdf", record.Retrieve(), 0644)
		if err != nil {
			return err
		}

		return nil
	})
}

File: java/src/main/java/com/bloock/quickstart/samples/builder/FromFile.java
package com.bloock.quickstart.samples.builder;

import com.bloock.quickstart.utils.Config;
import com.bloock.quickstart.utils.Logger;
import com.bloock.quickstart.utils.Sample;
import com.bloock.sdk.builder.Builder;
import com.bloock.sdk.entity.Record;
import java.nio.file.Files;
import java.nio.file.Paths;

public class FromFile extends Sample {
  public void run(Config config) throws Exception {
    byte[] file = Files.readAllBytes(Paths.get("../resources/dummy.pdf"));

    Record record = Builder.fromFile(file).build();

    Logger.success("Record was created successfully");

    String hash = record.getHash();

    assert hash.equals("7d87c5ea75f7378bb701e404c50639161af3eff66293e9f375b5f17eb50476f4");

    Logger.success("Hash: " + hash);

    // we can get the file back if needed
    Files.write(Paths.get("../resources/output.pdf"), record.retrieve());

    Logger.success("File written");
  }

  FromFile(String name) {
    super(name);
  }

  public static void main(String[] args) {
    new FromFile("FromFile");
  }
}
